plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -0,0 +1,266 @@
+            List<CompletableFuture<GetColumnStatisticsForTableResult>> getStatsFutures = columnChunks.stream()
+                    .map(partialColumns -> CompletableFuture.supplyAsync(() -> {
+                        GetColumnStatisticsForTableRequest request = new GetColumnStatisticsForTableRequest()
+                                .withCatalogId(catalogId)
+                                .withDatabaseName(table.getDatabaseName())
+                                .withTableName(table.getTableName())
+                                .withColumnNames(partialColumns);
+                        return glueClient.getColumnStatisticsForTable(request);
+                    }, readExecutor))
+                    .collect(toImmutableList());
+
+            HiveBasicStatistics tableStatistics = getHiveBasicStatistics(table.getParameters());
+            ImmutableMap.Builder<String, HiveColumnStatistics> columnStatsMapBuilder = ImmutableMap.builder();
+            for (CompletableFuture<GetColumnStatisticsForTableResult> future : getStatsFutures) {
+                GetColumnStatisticsForTableResult tableColumnsStats = getFutureValue(future, TrinoException.class);
+                for (ColumnStatistics columnStatistics : tableColumnsStats.getColumnStatisticsList()) {
+                    columnStatsMapBuilder.put(
+                            columnStatistics.getColumnName(),
+                            fromGlueColumnStatistics(columnStatistics.getStatisticsData(), tableStatistics.getRowCount()));
+                }
+            }
+            return columnStatsMapBuilder.build();
+        }
+        catch (AmazonServiceException ex) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+        }
+    }
+
+    @Override
+    public Map<String, HiveColumnStatistics> getPartitionColumnStatistics(Partition partition)
+    {
+        try {
+            List<List<Column>> columnChunks = Lists.partition(partition.getColumns(), GLUE_COLUMN_READ_STAT_PAGE_SIZE);
+            ImmutableList.Builder<CompletableFuture<GetColumnStatisticsForPartitionResult>> getStatsFutures = ImmutableList.builder();
+
+            columnChunks.forEach(partialColumns -> getStatsFutures.add(CompletableFuture.supplyAsync(() -> {
+                List<String> columnsNames = partialColumns.stream()
+                        .map(Column::getName)
+                        .collect(toImmutableList());
+
+                GetColumnStatisticsForPartitionRequest request = new GetColumnStatisticsForPartitionRequest()
+                        .withCatalogId(catalogId)
+                        .withDatabaseName(partition.getDatabaseName())
+                        .withTableName(partition.getTableName())
+                        .withColumnNames(columnsNames)
+                        .withPartitionValues(partition.getValues());
+
+                return glueClient.getColumnStatisticsForPartition(request);
+            }, readExecutor)));
```

Comment:
nit: 
```suggestion
            List<CompletableFuture<GetColumnStatisticsForPartitionResult>> getStatsFutures = columnChunks.stream()
                    .map(partialColumns -> CompletableFuture.supplyAsync(
                            () -> {
                                List<String> columnsNames = partialColumns.stream()
                                        .map(Column::getName)
                                        .collect(toImmutableList());

                                GetColumnStatisticsForPartitionRequest request = new GetColumnStatisticsForPartitionRequest()
                                        .withCatalogId(catalogId)
                                        .withDatabaseName(partition.getDatabaseName())
                                        .withTableName(partition.getTableName())
                                        .withColumnNames(columnsNames)
                                        .withPartitionValues(partition.getValues());
                                return glueClient.getColumnStatisticsForPartition(request);
                            }, readExecutor))
                    .collect(toImmutableList());
```

@hashhar which one looks nicer to you?

Reply:
Oh I see that you go this appraach in `getTableColumnStatistics`. Please be consistent.

Reply:
Still not consistent. `getPartitionColumnStatistics` uses `forEach ... getStatsFutures.add(`. And `getTableColumnStatistics` uses `.collect(toImmutableList());`. Please migrate to `.collect(toImmutableList());` with both.

Reply:
Sorry you're right !